Queries fail when using native text file reader on S3 objects with empty Content-Type header

Since Trino version 409, a previously working query has started to fail with `java.lang.IllegalArgumentException: Could not parse ''` when using the native text file reader.

The query is against a Glue table containing AWS ALB logs. The log files are gzip compressed and get delivered to S3 with an empty Content-Type header.

When `text-file.native-reader.enabled=false`, or with Trino version <=408, there query completes as expected.

When `text-file.native-reader.enabled=true` (the default since 409) then queries fail with this stacktrace:

```
io.trino.spi.TrinoException: Error opening Hive split s3://bucket-name/prefix/my_cool_file.log.gz (offset=0, length=322): Could not parse ''
	at io.trino.plugin.hive.line.LinePageSourceFactory.createPageSource(LinePageSourceFactory.java:170)
	at io.trino.plugin.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:216)
	at io.trino.plugin.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:154)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:49)
	at io.trino.split.PageSourceManager.createPageSource(PageSourceManager.java:62)
	at io.trino.operator.TableScanOperator.getOutput(TableScanOperator.java:298)
	at io.trino.operator.Driver.processInternal(Driver.java:396)
	at io.trino.operator.Driver.lambda$process$8(Driver.java:299)
	at io.trino.operator.Driver.tryWithLock(Driver.java:691)
	at io.trino.operator.Driver.process(Driver.java:291)
	at io.trino.operator.Driver.processForDuration(Driver.java:262)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:773)
	at io.trino.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:165)
	at io.trino.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:523)
	at io.trino.$gen.Trino_410____20230314_125440_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: Could not parse ''
	at com.google.common.net.MediaType.parse(MediaType.java:1078)
	at io.trino.plugin.hive.s3.TrinoS3FileSystem.getFileStatus(TrinoS3FileSystem.java:499)
	at org.apache.hadoop.fs.FilterFileSystem.getFileStatus(FilterFileSystem.java:454)
	at io.trino.filesystem.hdfs.HdfsInputFile.lambda$lazyStatus$2(HdfsInputFile.java:116)
	at io.trino.hdfs.authentication.NoHdfsAuthentication.doAs(NoHdfsAuthentication.java:25)
	at io.trino.hdfs.HdfsEnvironment.doAs(HdfsEnvironment.java:93)
	at io.trino.filesystem.hdfs.HdfsInputFile.lazyStatus(HdfsInputFile.java:116)
	at io.trino.filesystem.hdfs.HdfsInputFile.length(HdfsInputFile.java:72)
	at io.trino.plugin.hive.MonitoredTrinoInputFile.length(MonitoredTrinoInputFile.java:55)
	at io.trino.plugin.hive.line.LinePageSourceFactory.createPageSource(LinePageSourceFactory.java:151)
	... 17 more
Caused by: java.lang.IllegalStateException: undefined
	at com.google.common.base.Preconditions.checkState(Preconditions.java:486)
	at com.google.common.net.MediaType$Tokenizer.consumeTokenIfPresent(MediaType.java:1092)
	at com.google.common.net.MediaType$Tokenizer.consumeToken(MediaType.java:1100)
	at com.google.common.net.MediaType.parse(MediaType.java:1047)
	... 26 more
```

--- a/lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java
+++ b/lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java
@@ -517,11 +517,21 @@ public class TrinoS3FileSystem
         return new FileStatus(
                 getObjectSize(path, metadata),
                 // Some directories (e.g. uploaded through S3 GUI) return a charset in the Content-Type header
-                MediaType.parse(metadata.getContentType()).is(DIRECTORY_MEDIA_TYPE),
+                isDirectoryMediaType(nullToEmpty(metadata.getContentType())),
                 1,
                 BLOCK_SIZE.toBytes(),
                 lastModifiedTime(metadata),
                 qualifiedPath(path));
+    }
+
+    private static boolean isDirectoryMediaType(String contentType)
+    {
+        try {
+            return MediaType.parse(contentType).is(DIRECTORY_MEDIA_TYPE);
+        }
+        catch (IllegalArgumentException e) {
+            return false;
+        }
     }
 
     private long getObjectSize(Path path, ObjectMetadata metadata)

